homework/src/index.js

```diff
@@ -31,14 +34,83 @@
   }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
-    });
+    fetchJSON(url)
+      .then(data => {
+        const root = document.getElementById('root');
+        const repositoryDiv = createAndAppend('div', root, { class: 'repositoryDiv' });
+        const header = createAndAppend('header', repositoryDiv, { class: 'topping' });
+        createAndAppend('h1', header, { html: 'HYF Repositories ', class: 'hyfRepo' });
+        const selectBox = createAndAppend('select', header, { class: 'select-box' });
+        data.sort(function (a, b) { return a.name.localeCompare(b.name); });
+        for (let i = 0; i < data.length; i++) {
+          createAndAppend('option', selectBox, { value: i, html: data[i].name });
+        }
+        const infoBox = createAndAppend('div', root, { class: 'infoBox' });
+        const ContributorBox = createAndAppend('div', root, { class: 'ContributorBox' });
+        selectBox.addEventListener('change', () => {
+          const value = data[selectBox.selectedIndex].url;
+          fetchJSON(value)
+            .then(data => {
+              repositoryInfo(infoBox, data, value);
+            });
+          const contributors = data[selectBox.selectedIndex].contributors_url;
+          fetchJSON(contributors)
+            .then(data => {
+              ContributorsInfo(ContributorBox, data, contributors);
+            });
+        });
+        fetchJSON(data[0].url)
+          .then(data => {
+            repositoryInfo(infoBox, data);
+          });
+        fetchJSON(data[0].contributors_url)
+          .then(data => {
+            ContributorsInfo(ContributorBox, data);
+          });
+      })
+      .catch(err => {
+        const root = document.getElementById('root');
+        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
+
+      });
+  }
+
+  //adding rows function
+  function addRow(label, value, tbody) {
+    const row = createAndAppend('tr', tbody);
+    createAndAppend('td', row, { html: label });
+    createAndAppend('td', row, { html: value });
+  }
+
+  //information box of each repository
+  function repositoryInfo(infoBox, data) {
+    infoBox.innerHTML = '';
+    const table = createAndAppend('table', infoBox, { class: 'infoTable' });
+    const tbody = createAndAppend('tbody', table);
+    addRow('Repository', data.name, tbody);
```

Comment:
The repository name is supposed to be clickable link that takes you to the GitHub page of the selected repository.